django/db/backends/postgresql/base.py

```diff
@@ -227,12 +231,32 @@ def create_cursor(self, name=None):
         cursor.tzinfo_factory = utc_tzinfo_factory if settings.USE_TZ else None
         return cursor
 
+    @async_unsafe
     def chunked_cursor(self):
         self._named_cursor_idx += 1
+        # Get the current async task
+        # Note that right now this is behind @async_unsafe, so this is
+        # unreachable, but in future we'll start loosening this restriction.
+        # For now, it's here so that every use of "threading" is also async-compatible.
+        try:
+            if hasattr(asyncio, "current_task"):
+                # Python 3.7 and up
+                current_task = asyncio.current_task()
+            else:
+                # Python 3.6
+                current_task = asyncio.Task.current_task()
+        except RuntimeError:
+            current_task = None
+        if current_task:
+            task_ident = str(id(current_task))
+        else:
+            task_ident = "sync"
```

Comment:
Unfortunately I believe `asyncio.current_task` can return `None` _as well as_ raising a `RuntimeError`.

Reply:
I was afraid that this can be the case :disappointed:, so maybe ternary :wink: 
```python
task_ident = str(id(current_task)) if current_task else 'sync'
```

Reply:
Oh, interesting, I dislike ternary clauses (I think full `if` statements are easier to read). I did add a comment to make it clearer - can I convince you to leave it as an `if`? :)

Reply:
Sure, we can leave it :+1: